src/workerd/server/server.c++

```diff
@@ -1982,11 +1982,11 @@ public:
             // work for local development we need to pass an event type.
             static constexpr uint16_t hibernationEventTypeId = 8;
 
-            actorContainer->actor.emplace(
-                kj::refcounted<Worker::Actor>(*service.worker, actorContainer->getTracker(),
-                    kj::str(idPtr), true, kj::mv(makeActorCache), className, kj::mv(makeStorage),
-                    lock, kj::mv(loopback), timerChannel, kj::refcounted<ActorObserver>(),
-                    actorContainer->tryGetManagerRef(), hibernationEventTypeId));
+            actorContainer->actor.emplace(kj::refcounted<Worker::Actor>(*service.worker,
+                actorContainer->getTracker(), kj::str(idPtr), true, kj::mv(makeActorCache),
+                className, kj::mv(makeStorage), lock, kj::mv(loopback), timerChannel,
+                kj::refcounted<ActorObserver>(), kj::Own<SqliteObserver>(),
```

Comment:
`kj::Own<SqliteObserver>()` here constructs a null pointer, which will presumably lead to a segfault when accessed later.

Given that the implementation ultimately has a single class implementing both `ActorObserver` and `SqliteObserver`, perhaps `ActorObserver` itself should be declared to be a subclass of `SqliteObserver` so that it doesn't need to be passed separately here?

Reply:
That sounds fair